src/utils/module.c

```diff
@@ -100,21 +101,24 @@ gboolean bd_utils_have_kernel_module (const gchar *module_name, GError **error)
     }
     set_kmod_logging (ctx);
 
-    ret = kmod_module_new_from_name (ctx, module_name, &mod);
+    ret = kmod_module_new_from_lookup (ctx, module_name, &list);
     if (ret < 0) {
         g_set_error (error, BD_UTILS_MODULE_ERROR, BD_UTILS_MODULE_ERROR_FAIL,
                      "Failed to get the module: %s", strerror_l (-ret, c_locale));
         kmod_unref (ctx);
+        kmod_module_unref_list (list);
         freelocale (c_locale);
         return FALSE;
     }
 
+    mod = kmod_module_get_module (list);
```

Comment:
Isn't it better to iterate through the list in case more entries are returned? See https://github.com/lucasdemarchi/kmod/blob/fa67110f896cdef67f42cbc2206ae2a8524acee6/libkmod/README#L53

Reply:
And do we really need to check all possible matching modules? Isn't it enough for us that to know that there is at least one module matching the name/alias?

Reply:
Yeah, that crossed my mind as well when having dinner yesterday. I guess any first match might be enough in this case.